### ISSUE TITLE
Hotkeys

### ISSUE THREAD

**Issue by BrunIF** [wavexx/screenkey#25]
Can You add hotkeys to the program. For example,
_Ctrl + Super + S_ - Show/Stop show Screenkey
_Ctrl + Super + M_ - Show/Stop show Modifiers only

**Comment by wavexx** [comment]
The first is already available through Ctrl + Ctrl (using both control keys).

**Comment by BrunIF** [comment]
This keys cmbination not work on my Elementary OS  (Based on Ubuntu 14.04 LTS). 
Screenkey 0.8.1
I installed from this [instruction](http://www.webupd8.org/2015/11/display-keystrokes-in-your-screencasts.html) and their repository.

**Comment by wavexx** [comment]
On 27/01/16 21:04, Igor Bronovskyi wrote:

> This keys cmbination not work on my Elementary OS (Based on Ubuntu 14.04
> LTS).

Can you start screenkey in a terminal with debugging?

  screenkey -d

then immediately press both control keys.
It should show the following:

  % screenkey -d
  DEBUG:Screenkey:Options loaded.
  DEBUG:Screenkey:Restarting LabelManager.
  DEBUG:Screenkey:Thread started.
  DEBUG:Screenkey:Using StatusIcon.
  DEBUG:Screenkey:Key pressed  65507(ks): None (Control_L, mask: 00000000)
  DEBUG:Screenkey:Key released 65507(ks): Control_L
  DEBUG:Screenkey:Key pressed  65507(ks): None (Control_L, mask: 00000000)
  DEBUG:Screenkey:Key pressed  65508(ks): None (Control_R, mask: 00000100)
  INFO:Screenkey:Ctrl+Ctrl detected: screenkey disabled.

**Comment by BrunIF** [comment]
```
brun@pro-book:~$ screenkey -d
Gtk-Message: Failed to load module "pantheon-filechooser-module"
Gtk-Message: Failed to load module "canberra-gtk-module"
DEBUG:Screenkey:Options loaded.
DEBUG:Screenkey:Restarting LabelManager.
DEBUG:Screenkey:Thread started.
DEBUG:Screenkey:Restarting LabelManager.
DEBUG:Screenkey:Thread stopped.
DEBUG:Screenkey:Thread started.
DEBUG:Screenkey:Visible Shift changed: True.
DEBUG:Screenkey:Using AppIndicator.
DEBUG:Screenkey:Key filtered 65507(ks): Control_L
DEBUG:Screenkey:Key repeated 65507(ks): None (Control_L, mask: 00000000)
DEBUG:Screenkey:Key filtered 65508(ks): Control_R
INFO:Screenkey:Ctrl+Ctrl detected: screenkey disabled.
DEBUG:Screenkey:Key repeated 65508(ks): None (Control_R, mask: 00000100)
INFO:Screenkey:Ctrl+Ctrl detected: screenkey enabled.
DEBUG:Screenkey:Key released 65508(ks): Control_R
DEBUG:Screenkey:Key released 65508(ks): Control_R
DEBUG:Screenkey:Key released 65507(ks): Control_L
```

**Comment by wavexx** [comment]
Looks like key repeat is toggling recording on-off.
It's the first time I see key repeat for meta keys though.

Are you using something else than a regular keyboard?

**Comment by BrunIF** [comment]
Not working.
How bind another key to this? And bind keys with other actions?

**Comment by wavexx** [comment]
On 27/01/16 23:15, Igor Bronovskyi wrote:

> Not working.

I was asking if you're doing something special with the keyboard that
you're using.

It's not usual to have key-repeats for control keys.

> How bind another key to this? And bind keys with other actions?

It's currently not possible.

I'll have to implement some way to set bindings, but ctrl+ctrl is one of
the combinations which are likely to never cause trouble, but cannot be
bound the "usual" way either.

**Comment by wavexx** [comment]
On 27/01/16 23:15, Igor Bronovskyi wrote:

> Not working.

Can you try on the current master branch?

**Comment by BrunIF** [comment]
```
Installed /usr/local/lib/python2.7/dist-packages/screenkey-0.8.1-py2.7.egg
Processing dependencies for screenkey==0.8.1
Finished processing dependencies for screenkey==0.8.1
```

This key combination I have not used in the system, but not working. Nothing happens.

**Comment by wavexx** [comment]
On 31/01/16 12:52, Igor Bronovskyi wrote:

> |Installed
> /usr/local/lib/python2.7/dist-packages/screenkey-0.8.1-py2.7.egg
> Processing dependencies for screenkey==0.8.1 Finished processing
> dependencies for screenkey==0.8.1 |
> 
> This key combination I have not used in the system, but not working.
> Nothing happens.

Can you post again the log you get by running screenkey -d and pressing
both control keys?

**Comment by BrunIF** [comment]
```
brun@pro-book:~$ screenkey -d 
Gtk-Message: Failed to load module "pantheon-filechooser-module"
Gtk-Message: Failed to load module "canberra-gtk-module"
DEBUG:Screenkey:Options loaded.
DEBUG:Screenkey:Restarting LabelManager.
DEBUG:Screenkey:Thread started.
DEBUG:Screenkey:Restarting LabelManager.
DEBUG:Screenkey:Thread stopped.
DEBUG:Screenkey:Thread started.
DEBUG:Screenkey:Visible Shift changed: True.
DEBUG:Screenkey:Using AppIndicator.
DEBUG:Screenkey:Key filtered 65507(ks): Control_L
DEBUG:Screenkey:Key repeated 65507(ks): None (Control_L, mask: 00010000)
DEBUG:Screenkey:Key filtered 65508(ks): Control_R
DEBUG:Screenkey:Key repeated 65508(ks): None (Control_R, mask: 00010100)
DEBUG:Screenkey:Key released 65508(ks): Control_R
DEBUG:Screenkey:Key released 65508(ks): Control_R
DEBUG:Screenkey:Key released 65507(ks): Control_L
DEBUG:Screenkey:Key released 65507(ks): Control_L
```

**Comment by wavexx** [comment]
I cannot reproduce this.

Are you using some input methods, such as ibus or fcitx?
What method did you set? (can you try to disable it to test?).

**Comment by BrunIF** [comment]
Very fine work on **Ubuntu** but not work on **Elementary OS 0.3**
I don't understand why.

**Comment by wavexx** [comment]
On 03/02/16 23:18, Igor Bronovskyi wrote:

> Very fine work on _Ubuntu_ but not work on _Elementary OS 0.3_
> I don't understand why.

I suspect there's a different input method or it's configured differently.

Unfortunately I don't have time to try elementary os to find out.
Any help in debugging would be appreciated.

**Comment by xintron** [comment]
I don't have the same problem but I do request the HotKey support for "toggle enabled". The problem I'm faced with is only one ctrl key mapped on my laptop keyboard (Macbook running Linux). While it's possible to temporarily map Control_R to another button it would be even better to allow they combination to be set (my personal idea is to use Ctrl+Esc).

**Comment by wavexx** [comment]
On Thu, Feb 18 2016, Marcus Carlsson notifications@github.com wrote:

> I don't have the same problem but I do request the HotKey support for "toggle
> enabled". The problem I'm faced with is only one ctrl key mapped on my laptop
> keyboard (Macbook running Linux). While it's possible to temporarily map
> Control_R to another button it would be even better to allow they combination
> to be set (my personal idea is to use Ctrl+Esc).

This is definitely something that needs to be done, no discussion.

I was looking for a widget already designed to capture new keybindings
("accelerators" in gtk), but I came empty handed. It's annoying that one
has to implement this.

But on the other hand, it's probably unlikely that existing widgets can
capture stuff as Ctrl+Ctrl.

I think I'll add a text input where you _write_ the keyboard sequence to
be captured (as "Ctrl+Ctrl", "Ctrl+ESC", etc).

**Comment by uli-heller** [comment]
I do have a similar issue. Ctrl-Ctrl doesn''t work for me. It is always detected twice:

```
DEBUG:Screenkey:Key filtered 65508(ks): Control_R
INFO:Screenkey:Ctrl+Ctrl detected: screenkey disabled.
DEBUG:Screenkey:Key repeated 65508(ks): None (Control_R, mask: 00000100)
INFO:Screenkey:Ctrl+Ctrl detected: screenkey enabled.
DEBUG:Screenkey:Key released 65508(ks): Control_R
```

Once, I've been able to produce two lines in between the detections:

```
DEBUG:Screenkey:Key filtered 65508(ks): Control_R
INFO:Screenkey:Ctrl+Ctrl detected: screenkey disabled.
DEBUG:Screenkey:Key pressed  65507(ks): None (Control_L, mask: 00000000)
DEBUG:Screenkey:Key pressed  65508(ks): None (Control_R, mask: 00000100)
INFO:Screenkey:Ctrl+Ctrl detected: screenkey enabled.
DEBUG:Screenkey:Key released 65508(ks): Control_R
```

My environment:
- Ubuntu-14.04
- Cinnamon desktop
- screenkey-0.8.1

I'll try it with an Ubuntu-Mate-16.04 later on.

**Comment by wavexx** [comment]
Although I didn't do much yet to improve the hotkey, the current version in master should at least work-around the double detection. Could you try?

**Comment by uli-heller** [comment]
Current master works fine with me, congratulations and thanks!

**Comment by wavexx** [comment]
In a similar spirit, I now allow also shift+shift and alt+alt for the same effect.
Again, this is still not a proper configurable hotkey (so I'm not closing this), but should help in some keyboard layouts.

**Comment by wavexx** [comment]
The advantage here is that these combinations cannot be bound as hotkeys at all, so they're free from conflicting from a regular hotkey in any other program.

**Comment by pataquets** [comment]
Same problem here. Dell laptop running Precise + Gnome classic. Tried holding/releasing the keys in different orderings, but no change.
Debug output, in case it is of any help:

```
$ screenkey --no-detach --debug
DEBUG:Screenkey:file /home/amontero/.cache/screenkey.dat does not exists.
DEBUG:Screenkey:Thread created
Xlib.protocol.request.QueryExtension
Xlib.protocol.request.QueryExtension
DEBUG:Screenkey:Thread started.
DEBUG:Screenkey:Using AppIndicator.
DEBUG:Screenkey:Key 37(keycode) pressed. Symbols [u'\uffe3', '', '', '']
DEBUG:Screenkey:Key 105(keycode) pressed. Symbols [u'\uffe4', '', '', '']
DEBUG:Screenkey:Key 105(keycode) released. Symbols [u'\uffe4', '', '', '']
DEBUG:Screenkey:Key 37(keycode) released. Symbols [u'\uffe3', '', '', '']
DEBUG:Screenkey:Key 50(keycode) pressed. Symbols [u'\uffe1', '', '', '']
DEBUG:Screenkey:Key 62(keycode) pressed. Symbols [u'\uffe2', '', '', '']
DEBUG:Screenkey:Key 62(keycode) released. Symbols [u'\uffe2', '', '', '']
DEBUG:Screenkey:Key 50(keycode) released. Symbols [u'\uffe1', '', '', '']
DEBUG:Screenkey:Key 64(keycode) pressed. Symbols [u'\uffe9', u'\uffe7', '', '']
DEBUG:Screenkey:Key 108(keycode) pressed. Symbols [u'\ufe03', '', '', '']
DEBUG:Screenkey:Key 64(keycode) released. Symbols [u'\uffe9', u'\uffe7', '', '']
DEBUG:Screenkey:Key 108(keycode) released. Symbols [u'\ufe03', '', '', '']
```

**Comment by wavexx** [comment]
On Tue, Sep 27 2016, Alfonso Montero notifications@github.com wrote:

> Same problem here. Dell laptop running Precise + Gnome classic. Tried holding/releasing the keys in different orderings, but no change.
> Debug output, in case it is of any help:

By looking the log, you're using the original screenkey, not this fork.

Note that the ubuntu (and debian) repository still has the original
fork. See http://www.webupd8.org/2016/05/tool-to-display-keystrokes-in.html if you
want a pre-packaged version.

**Comment by pozitron57** [comment]
@wavexx, it would be great if ctrl+ctrl hotkey will be mentioned in the readme on the main page! I'm quite surprised to find it here but not there.

**Comment by wavexx** [comment]
On Tue, Apr 25 2017, Sergey Lisakov wrote:
> @wavexx, it would be great if ctrl+ctrl hotkey will be mentioned in the readme
> on the main page! I'm quite surprised to find it here but not there.

It's there:

https://github.com/wavexx/screenkey/#controlling-visibility

But maybe it's not clear enough?
How/where would you place it?

**Comment by pozitron57** [comment]
@wavexx, it seems that I'm completely dumb. It is mentioned just in the right place. Maybe just one can give a little more visibility using `<kbd></kbd>` tag like so:

>To disable screenkey while recording (for example, during password prompts), press both control keys (<kbd>Ctrl</kbd> + <kbd>Ctrl</kbd>), or both shift keys (<kbd>Shift</kbd> + <kbd>Shift</kbd>), or both alt keys (<kbd>Alt</kbd> + <kbd>Alt</kbd>) at the same time.

It may seem to be a little excessive, but would help someone not very patient who just scrolls the page back and forth looking for the shotcuts. Well, I must admit that this type of behavior is not very smart but rather common. Thanks for the great tool!

**Comment by wavexx** [comment]
On Tue, Apr 25 2017, Sergey Lisakov wrote:
>>To disable screenkey while recording (for example, during password
>>prompts), press both control keys (<kbd>Ctrl</kbd> + <kbd>Ctrl</kbd>),
>>or both shift keys (<kbd>Shift</kbd> + <kbd>Shift</kbd>), or both alt
>>keys (<kbd>Alt</kbd> + <kbd>Alt</kbd>) at the same time.

I'll do that.

> It may seem to be a little excessive, but would help someone not very
> patient who just scrolls the page back and forth looking for the
> shotcuts. Well, I must admit that this type of behavior is not very
> smart but rather common. Thanks for the great tool!

Screenkey lacks some help function as well.

If you start screenkey from a package (debian/arch/etc), you'll likely
never see the README, and there's no mention of this in the gui.

For a teaching tool, it's not a great start ;)

**Comment by DanitiKain** [comment]
screenkey -d
DEBUG:Screenkey:Options loaded.
DEBUG:Screenkey:Restarting LabelManager.
DEBUG:Screenkey:Thread started.
DEBUG:Screenkey:Restarting LabelManager.
DEBUG:Screenkey:Thread stopped.
DEBUG:Screenkey:Thread started.
DEBUG:Screenkey:Visible Shift changed: True.
DEBUG:Screenkey:Restarting LabelManager.
DEBUG:Screenkey:Thread stopped.
DEBUG:Screenkey:Thread started.
DEBUG:Screenkey:Show Whitespace changed: True.
DEBUG:Screenkey:Using StatusIcon.
DEBUG:Screenkey:Key pressed  65508(ks): None (Control_R, mask: 00000000)
DEBUG:Screenkey:Key released 65508(ks): Control_R
DEBUG:Screenkey:Key pressed  269025067(ks): None (XF86WakeUp, mask: 00000000)
DEBUG:Screenkey:Label updated: u'\u200c<u>XF86WakeUp</u>'.
DEBUG:Screenkey:Key released 269025067(ks): XF86WakeUp
DEBUG:Screenkey:Key pressed  65505(ks): None (Shift_L, mask: 00000000)
DEBUG:Screenkey:Key pressed  65506(ks): None (Shift_R, mask: 00000001)
DEBUG:Screenkey:Key released 65506(ks): Shift_R
DEBUG:Screenkey:Key released 65505(ks): Shift_L
fDEBUG:Screenkey:Key pressed    102(ks): 'f' (f, mask: 00000000)
DEBUG:Screenkey:Label updated: u'\u200c<u>f</u>'.
jDEBUG:Screenkey:Key pressed    106(ks): 'j' (j, mask: 00000000)
DEBUG:Screenkey:Label updated: u'\u200c<u>f\u200cj</u>'.
DEBUG:Screenkey:Key released   102(ks): f
DEBUG:Screenkey:Key released   106(ks): j
DEBUG:Screenkey:Key pressed  65513(ks): None (Alt_L, mask: 00000000)
DEBUG:Screenkey:Key pressed  65514(ks): None (Alt_R, mask: 00001000)
DEBUG:Screenkey:Key released 65514(ks): Alt_R
DEBUG:Screenkey:Key released 65513(ks): Alt_L
sDEBUG:Screenkey:Key pressed    115(ks): 's' (s, mask: 00000000)
DEBUG:Screenkey:Label updated: u'\u200c<u>s</u>'.
tDEBUG:Screenkey:Key pressed    116(ks): 't' (t, mask: 00000000)
DEBUG:Screenkey:Label updated: u'\u200c<u>s\u200ct</u>'.
iDEBUG:Screenkey:Key pressed    105(ks): 'i' (i, mask: 00000000)
DEBUG:Screenkey:Label updated: u'\u200cs\u200c<u>t\u200ci</u>'.
DEBUG:Screenkey:Key released   115(ks): s
DEBUG:Screenkey:Key released   116(ks): t
DEBUG:Screenkey:Key released   105(ks): i
lDEBUG:Screenkey:Key pressed    108(ks): 'l' (l, mask: 00000000)
DEBUG:Screenkey:Label updated: u'\u200cs\u200ct\u200ci\u200c<u>l</u>'.
DEBUG:Screenkey:Key released   108(ks): l
lDEBUG:Screenkey:Key pressed    108(ks): 'l' (l, mask: 00000000)
DEBUG:Screenkey:Label updated: u'\u200cs\u200ct\u200ci\u200cl\u200c<u>l</u>'.
DEBUG:Screenkey:Key released   108(ks): l
 DEBUG:Screenkey:Key pressed     32(ks): ' ' (space, mask: 00000000)
DEBUG:Screenkey:Label updated: u'\u200cs\u200ct\u200ci\u200cl\u200cl\u200c<u>\u2423</u>'.
DEBUG:Screenkey:Key released    32(ks): space
oDEBUG:Screenkey:Key pressed    111(ks): 'o' (o, mask: 00000000)
DEBUG:Screenkey:Label updated: u'\u200cs\u200ct\u200ci\u200cl\u200cl\u200c\u2423\u200c<u>o</u>'.
nDEBUG:Screenkey:Key pressed    110(ks): 'n' (n, mask: 00000000)
DEBUG:Screenkey:Label updated: u'\u200cs\u200ct\u200ci\u200cl\u200cl\u200c\u2423\u200c<u>o\u200cn</u>'.
DEBUG:Screenkey:Key released   111(ks): o
DEBUG:Screenkey:Key released   110(ks): n
yDEBUG:Screenkey:Key pressed    121(ks): 'y' (y, mask: 00000000)
DEBUG:Screenkey:Label updated: u'\u200cs\u200ct\u200ci\u200cl\u200cl\u200c\u2423\u200co\u200cn\u200c<u>y</u>'.
DEBUG:Screenkey:Key released   121(ks): y
DEBUG:Screenkey:Key pressed  65288(ks): u'\x08' (BackSpace, mask: 00000000)
DEBUG:Screenkey:Label updated: u'\u200cs\u200ct\u200ci\u200cl\u200cl\u200c\u2423\u200co\u200cn'.
DEBUG:Screenkey:Key released 65288(ks): BackSpace


Modifier keys to turn this off are not working.  Tried all the combinations ctrl+ctrl, alt+alt and shift+shift.

This is in the latest version of Antergos and screenkey install from AUR.

**Comment by DanitiKain** [comment]
I cloned the current repo and ran debug.  Here is the output.  It didn't work either.

screenkey -d  
DEBUG:Screenkey:Options loaded.
DEBUG:Screenkey:Restarting LabelManager.
DEBUG:Screenkey:Thread started.
DEBUG:Screenkey:Restarting LabelManager.
DEBUG:Screenkey:Thread stopped.
DEBUG:Screenkey:Thread started.
DEBUG:Screenkey:Visible Shift changed: True.
DEBUG:Screenkey:Restarting LabelManager.
DEBUG:Screenkey:Thread stopped.
DEBUG:Screenkey:Thread started.
DEBUG:Screenkey:Show Whitespace changed: True.
DEBUG:Screenkey:Using StatusIcon.
tDEBUG:Screenkey:Key pressed    116(ks): 't' (t, mask: 00000000)
DEBUG:Screenkey:Label updated: u'\u200c<u>t</u>'.
hDEBUG:Screenkey:Key pressed    104(ks): 'h' (h, mask: 00000000)
DEBUG:Screenkey:Label updated: u'\u200c<u>t\u200ch</u>'.
DEBUG:Screenkey:Key released   116(ks): t
iDEBUG:Screenkey:Key pressed    105(ks): 'i' (i, mask: 00000000)
DEBUG:Screenkey:Label updated: u'\u200ct\u200ch\u200c<u>i</u>'.
DEBUG:Screenkey:Key released   104(ks): h
DEBUG:Screenkey:Key released   105(ks): i
sDEBUG:Screenkey:Key pressed    115(ks): 's' (s, mask: 00000000)
DEBUG:Screenkey:Label updated: u'\u200ct\u200ch\u200ci\u200c<u>s</u>'.
DEBUG:Screenkey:Key released   115(ks): s
 DEBUG:Screenkey:Key pressed     32(ks): ' ' (space, mask: 00000000)
DEBUG:Screenkey:Label updated: u'\u200ct\u200ch\u200ci\u200cs\u200c<u>\u2423</u>'.
DEBUG:Screenkey:Key released    32(ks): space
sDEBUG:Screenkey:Key pressed    115(ks): 's' (s, mask: 00000000)
DEBUG:Screenkey:Label updated: u'\u200ct\u200ch\u200ci\u200cs\u200c\u2423\u200c<u>s</u>'.
DEBUG:Screenkey:Key released   115(ks): s
DEBUG:Screenkey:Key pressed  65288(ks): u'\x08' (BackSpace, mask: 00000000)
DEBUG:Screenkey:Label updated: u'\u200ct\u200ch\u200ci\u200cs\u200c\u2423'.
DEBUG:Screenkey:Key released 65288(ks): BackSpace
iDEBUG:Screenkey:Key pressed    105(ks): 'i' (i, mask: 00000000)
DEBUG:Screenkey:Label updated: u'\u200ct\u200ch\u200ci\u200cs\u200c\u2423\u200c<u>i</u>'.
DEBUG:Screenkey:Key released   105(ks): i
sDEBUG:Screenkey:Key pressed    115(ks): 's' (s, mask: 00000000)
DEBUG:Screenkey:Label updated: u'\u200ct\u200ch\u200ci\u200cs\u200c\u2423\u200ci\u200c<u>s</u>'.
DEBUG:Screenkey:Key released   115(ks): s
 DEBUG:Screenkey:Key pressed     32(ks): ' ' (space, mask: 00000000)
DEBUG:Screenkey:Label updated: u'\u200ct\u200ch\u200ci\u200cs\u200c\u2423\u200ci\u200cs\u200c<u>\u2423</u>'.
DEBUG:Screenkey:Key released    32(ks): space
aDEBUG:Screenkey:Key pressed     97(ks): 'a' (a, mask: 00000000)
DEBUG:Screenkey:Label updated: u'\u200ct\u200ch\u200ci\u200cs\u200c\u2423\u200ci\u200cs\u200c\u2423\u200c<u>a</u>'.
 DEBUG:Screenkey:Key pressed     32(ks): ' ' (space, mask: 00000000)
DEBUG:Screenkey:Label updated: u'\u200ct\u200ch\u200ci\u200cs\u200c\u2423\u200ci\u200cs\u200c\u2423\u200c<u>a\u200c\u2423</u>'.
DEBUG:Screenkey:Key released    97(ks): a
DEBUG:Screenkey:Key released    32(ks): space
tDEBUG:Screenkey:Key pressed    116(ks): 't' (t, mask: 00000000)
DEBUG:Screenkey:Label updated: u'\u200ct\u200ch\u200ci\u200cs\u200c\u2423\u200ci\u200cs\u200c\u2423\u200ca\u200c\u2423\u200c<u>t</u>'.
eDEBUG:Screenkey:Key pressed    101(ks): 'e' (e, mask: 00000000)
DEBUG:Screenkey:Label updated: u'\u200ct\u200ch\u200ci\u200cs\u200c\u2423\u200ci\u200cs\u200c\u2423\u200ca\u200c\u2423\u200c<u>t\u200ce</u>'.
DEBUG:Screenkey:Key released   116(ks): t
DEBUG:Screenkey:Key released   101(ks): e
sDEBUG:Screenkey:Key pressed    115(ks): 's' (s, mask: 00000000)
DEBUG:Screenkey:Label updated: u'\u200ct\u200ch\u200ci\u200cs\u200c\u2423\u200ci\u200cs\u200c\u2423\u200ca\u200c\u2423\u200ct\u200ce\u200c<u>s</u>'.
tDEBUG:Screenkey:Key pressed    116(ks): 't' (t, mask: 00000000)
DEBUG:Screenkey:Label updated: u'\u200ct\u200ch\u200ci\u200cs\u200c\u2423\u200ci\u200cs\u200c\u2423\u200ca\u200c\u2423\u200ct\u200ce\u200c<u>s\u200ct</u>'.
DEBUG:Screenkey:Key released   115(ks): s
DEBUG:Screenkey:Key released   116(ks): t
DEBUG:Screenkey:Key pressed  65508(ks): None (Control_R, mask: 00000000)
DEBUG:Screenkey:Key released 65508(ks): Control_R
DEBUG:Screenkey:Key pressed  269025067(ks): None (XF86WakeUp, mask: 00000000)
DEBUG:Screenkey:Label updated: u'\u200c<u>XF86WakeUp</u>'.
DEBUG:Screenkey:Key released 269025067(ks): XF86WakeUp
DEBUG:Screenkey:Key pressed  65513(ks): None (Alt_L, mask: 00000000)
DEBUG:Screenkey:Key pressed  65514(ks): None (Alt_R, mask: 00001000)
DEBUG:Screenkey:Key released 65514(ks): Alt_R
DEBUG:Screenkey:Key released 65513(ks): Alt_L
DEBUG:Screenkey:Key pressed  65505(ks): None (Shift_L, mask: 00000000)
DEBUG:Screenkey:Key pressed  65506(ks): None (Shift_R, mask: 00000001)
DEBUG:Screenkey:Key released 65506(ks): Shift_R
DEBUG:Screenkey:Key released 65505(ks): Shift_L
DEBUG:Screenkey:Key pressed  65505(ks): None (Shift_L, mask: 00000000)
DEBUG:Screenkey:Key pressed  65506(ks): None (Shift_R, mask: 00000001)
DEBUG:Screenkey:Key released 65506(ks): Shift_R
DEBUG:Screenkey:Key released 65505(ks): Shift_L
DEBUG:Screenkey:Key pressed  65506(ks): None (Shift_R, mask: 00000000)
DEBUG:Screenkey:Key pressed  65505(ks): None (Shift_L, mask: 00000001)
DEBUG:Screenkey:Key released 65506(ks): Shift_R
DEBUG:Screenkey:Key released 65505(ks): Shift_L
DEBUG:Screenkey:Key pressed  65514(ks): None (Alt_R, mask: 00000000)
DEBUG:Screenkey:Key pressed  65513(ks): None (Alt_L, mask: 00001000)
DEBUG:Screenkey:Key released 65514(ks): Alt_R
DEBUG:Screenkey:Key released 65513(ks): Alt_L
DEBUG:Screenkey:Key pressed  65508(ks): None (Control_R, mask: 00000000)
DEBUG:Screenkey:Key released 65508(ks): Control_R
DEBUG:Screenkey:Key pressed  269025067(ks): None (XF86WakeUp, mask: 00000000)
DEBUG:Screenkey:Label updated: u'\u200c<u>XF86WakeUp</u>'.
DEBUG:Screenkey:Key released 269025067(ks): XF86WakeUp
DEBUG:Screenkey:Key pressed  65514(ks): None (Alt_R, mask: 00000000)
DEBUG:Screenkey:Key pressed  65513(ks): None (Alt_L, mask: 00001000)
DEBUG:Screenkey:Key released 65514(ks): Alt_R
DEBUG:Screenkey:Key released 65513(ks): Alt_L
DEBUG:Screenkey:Key pressed  65505(ks): None (Shift_L, mask: 00000000)
DEBUG:Screenkey:Key pressed  65506(ks): None (Shift_R, mask: 00000001)
DEBUG:Screenkey:Key released 65506(ks): Shift_R
DEBUG:Screenkey:Key released 65505(ks): Shift_L
DEBUG:Screenkey:Key pressed  65507(ks): None (Control_L, mask: 00000000)
